pkg/frontend/querymiddleware/step_align.go

```diff
@@ -18,26 +18,32 @@ import (
 )
 
 type stepAlignMiddleware struct {
-	next    MetricsQueryHandler
-	limits  Limits
-	logger  log.Logger
-	aligned *prometheus.CounterVec
+	next       MetricsQueryHandler
+	limits     Limits
+	logger     log.Logger
+	notAligned prometheus.Counter
+	adjusted   *prometheus.CounterVec
 }
 
 // newStepAlignMiddleware creates a middleware that aligns the start and end of request to the step to
 // improve the cacheability of the query results based on per-tenant configuration.
 func newStepAlignMiddleware(limits Limits, logger log.Logger, registerer prometheus.Registerer) MetricsQueryMiddleware {
-	aligned := promauto.With(registerer).NewCounterVec(prometheus.CounterOpts{
-		Name: "cortex_query_frontend_queries_step_aligned_total",
```

Comment:
is this metric now gone? so you replaced the positive one with a negative one? wasn't clear from the description. Might make sense to mention it in a changelog entry too

Reply:
oh, it was renamed to `cortex_query_frontend_non_step_aligned_queries_adjusted_total`, not "reversed"